src/charm.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm for Vault running on Kubernetes.
+
+For more information on Vault, please visit https://www.vaultproject.io/.
+"""
+
+import json
+import logging
+import os
+
+from charms.tls_certificates_interface.v0.tls_certificates import (
+    Cert,
+    InsecureCertificatesProvides,
+)
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus
+from ops.pebble import Layer
+
+import interface_vault_operator_peers
+from vault import Vault
+
+logger = logging.getLogger(__name__)
+
+
+STORAGE_PATH = "/var/lib/juju/storage/vault-storage/0"
+
+
+class VaultCharm(CharmBase):
```

Comment:
While I appreciate this is early days, this looks really clean overall :)

Reply:
:)